src/pages/productpage/Product.jsx

```diff
@@ -1,8 +1,22 @@
 import "./Product.css";
+import { useEffect } from "react";
 import { ProductFilter, ProductCard } from "../../components";
 import { useProductContext } from "../../context/ProductContext";
+import { getAllProducts } from "../../api/apicall";
+
 const Product = () => {
-	const { filteredData } = useProductContext();
+	const { filteredData, productDispatch } = useProductContext();
+
+	useEffect(() => {
+		const getData = async () => {
+			const response = await getAllProducts();
+
+			response.success
+				? productDispatch({ type: "SET_PRODUCTS", payload: response.products })
+				: productDispatch({ type: "SET_PRODUCTS", payload: [] });
+		};
+		getData();
+	}, [productDispatch]);
```

Comment:
I think you can only give [] instead of [productDispatch]

Reply:
i did that previously. But it it giving a warning and causing the netlify deploy to fail